src/linter.js

```diff
@@ -465,78 +446,70 @@ export default class Linter {
     return true;
   }
 
-  scan(deps = {}) {
-    return this.extractMetadata(deps)
-      .then(() => {
-        return this.io.getFiles();
-      })
-      .then((files) => {
-        if (this.config.scanFile) {
-          if (!this.config.scanFile.some((f) => Object.keys(files).includes(f))) {
-            const _files = this.config.scanFile.join(', ');
-            throw new Error(`Selected file(s) not found: ${_files}`);
-          }
+  async scan(deps = {}) {
+    try {
+      await this.extractMetadata(deps);
+      const files = await this.io.getFiles();
+
+      if (this.config.scanFile &&
+        !this.config.scanFile.some((f) => Object.keys(files).includes(f))) {
```

Comment:
Nit, align `!this.config.scanFile...` with `this.config.scanFile` from the previous line.